clients/src/test/java/org/apache/kafka/common/message/MessageTest.java

```diff
@@ -419,10 +425,16 @@ public void testTxnOffsetCommitRequestVersions() throws Exception {
 
         for (short version = 0; version <= ApiKeys.TXN_OFFSET_COMMIT.latestVersion(); version++) {
             TxnOffsetCommitRequestData requestData = request.get();
-            if (version < 6) {
+            if (version < 2) {
                 requestData.topics().get(0).partitions().get(0).setCommittedLeaderEpoch(-1);
             }
 
+            if (version < 3) {
+                requestData.setGroupInstanceId(null);
+                requestData.setMemberId("");
```

Comment:
We couldn't use JoinGroupRequest/Response defined structs here due to import control